shell/directives/clean-tooltip.ts

```diff
@@ -0,0 +1,260 @@
+import { DirectiveBinding, Directive } from 'vue';
+import { destroyTooltip, createTooltip } from 'floating-vue';
+import { purifyHTML } from '@shell/plugins/clean-html';
+
+// This is a singleton tooltip implementation that improves performance on pages with many tooltips.
```

Comment:
This new singleton pattern isn't mentioned in the PR description, instead it looks like a performance improvement not related to warnings?

It feels like there's a lot of work / lines of code here when there should already be a way to do the same natively via floating-vue's popper